docs/source/ingest/destination_connectors/pinecone.rst

```diff
@@ -0,0 +1,74 @@
+Pinecone
+===========
+
+Batch process all your records using ``unstructured-ingest`` to store structured outputs and embeddings locally on your filesystem and upload those to a Pinecone index.
+
+First you'll need to install the Pinecone dependencies as shown here.
+
+.. code:: shell
+
+  pip install "unstructured[pinecone]"
+
+Run Locally
+-----------
+The upstream connector can be any of the ones supported, but for convenience here, showing a sample command using the
+upstream s3 connector. This will create new files on your local.
+
+.. tabs::
+
+   .. tab:: Shell
+
+      .. code:: shell
+
+        unstructured-ingest \
+            s3 \
+            --remote-url s3://utic-dev-tech-fixtures/small-pdf-set/ \
+            --anonymous \
+            --output-dir s3-small-batch-output \
+            --num-processes 2 \
+            --verbose \
+            --strategy fast \
+            --embedding-api-key <your openai api key here> \
+            pinecone \
+            --api-key <your pinecone api key here> \
+            --index-name <your index name here, ie. ingest-test> \
+            --environment <your environment name here, ie. gcp-starter>
+
+   .. tab:: Python
+
+      .. code:: python
+
+        import subprocess
+
+        command = [
+          "unstructured-ingest",
+          "s3",
+          "--remote-url", "s3://utic-dev-tech-fixtures/small-pdf-set/",
+          "--anonymous",
+          "--output-dir", "s3-small-batch-output \",
+          "--num-processes", "2",
+          "--verbose",
+          "--strategy", "fast",
+          "--embedding-api-key", "<your openai api key here>",
```

Comment:
not just openai
..also, shouldn't you also need to specify --embedding-provider?

Reply:
Agreed, addressed with [update docs](https://github.com/Unstructured-IO/unstructured/pull/1774/commits/80fed3bd90fd9a1a40f3d101e37da761ff6769ad) . Didn't include `--embedding-api-key` since it is not mandatory for all embedding providers, please let me know if we'd like that included.

Reply:
it looks like you updated in the shell script part of the doc, but not here in the python bit as well